src/meta/src/rpc/elections.rs

```diff
@@ -0,0 +1,838 @@
+    ) {
+        put_leader_info(leader, meta_store).await;
+        put_lease_info(lease, meta_store).await;
+    }
+
+    /// Default setup
+    /// ## Returns:
+    /// lease timeout, meta store, leader info, lease info, lease registration time
+    async fn default_setup() -> (u64, Arc<MemStore>, MetaLeaderInfo, MetaLeaseInfo, Duration) {
+        let lease_timeout = 10;
+        let mock_meta_store = Arc::new(MemStore::new());
+        let leader_info = MetaLeaderInfo {
+            node_address: "localhost:1234".into(),
+            lease_id: 123,
+        };
+        let now = since_epoch();
+        let lease_info = MetaLeaseInfo {
+            leader: Some(leader_info.clone()),
+            lease_register_time: now.as_secs(),
+            lease_expire_time: now.as_secs() + lease_timeout,
+        };
+        put_leader_lease(&leader_info, &lease_info, &mock_meta_store).await;
+        (lease_timeout, mock_meta_store, leader_info, lease_info, now)
+    }
+
+    #[tokio::test]
+    async fn test_manage_term() {
+        let mock_meta_store = Arc::new(MemStore::new());
+        let lease_timeout = 10;
+
+        // Leader: If nobody was elected leader renewing lease fails and leader is marked as failed
+        let leader_info = MetaLeaderInfo {
+            node_address: "localhost:1234".into(),
+            lease_id: 123,
+        };
+        assert!(
+            !manage_term(true, &leader_info, lease_timeout, &mock_meta_store)
+                .await
+                .unwrap()
+        );
+
+        // Follower: If nobody was elected leader renewing lease also fails
+        assert!(
+            !manage_term(false, &leader_info, lease_timeout, &mock_meta_store)
+                .await
+                .unwrap()
+        );
+    }
+
+    #[tokio::test]
+    async fn leader_should_renew_lease() {
+        // if node is leader lease should be renewed
+        let (lease_timeout, mock_meta_store, leader_info, _, _) = default_setup().await;
+        let now = since_epoch();
+        let lease_info = MetaLeaseInfo {
+            leader: Some(leader_info.clone()),
+            lease_register_time: now.as_secs(),
+            lease_expire_time: now.as_secs() + lease_timeout,
+        };
+        put_leader_lease(&leader_info, &lease_info, &mock_meta_store).await;
+        assert!(
+            manage_term(true, &leader_info, lease_timeout, &mock_meta_store)
+                .await
+                .unwrap(),
+            "Leader should still be in power after updating lease"
+        );
+        let (_, new_lease_info) = get_infos_obj(&mock_meta_store).await.unwrap();
+        assert_eq!(
+            now.as_secs() + lease_timeout,
+            new_lease_info.get_lease_expire_time(),
+            "Lease was not extended by {}s, but by {}s",
+            lease_timeout,
+            new_lease_info.get_lease_expire_time() - lease_info.get_lease_expire_time()
+        );
+    }
+
+    #[tokio::test]
+    async fn follower_cannot_renew_lease() {
```

Comment:
```suggestion
    async fn follower_cannot_renew_lease() -> Result<()> {
``
So that you don't have to call `unwrap()` every where and use `?` instead.

Reply:
I am getting an error when I am trying to return a Result in a test... 

```
the trait bound `bool: Termination` is not satisfied
required for `std::result::Result<bool, meta_store::MetaStoreError>` to implement `Termination`
```

Reply:
It should works and I tried locally. Uses `MetaResult` for all function in this crate if needed.